docs/examples/document_segmentation.md

```diff
@@ -0,0 +1,143 @@
+# Document Segmentation
```

Comment:
you'll have to add this in the mkdocs.yml file
or even reference it in examples/index.md